github-connector/internal/registration/registerservice.go

```diff
@@ -1,1020 +1,123 @@
-										"type": "object",
-										"properties": {
-											"href": {
-												"type": "string",
-												"example": "https://api.github.com/repos/Codertocat/Hello-World/pulls/comments{/number}"
-											}
-										}
-									},
-									"commits": {
-										"type": "object",
-										"properties": {
-											"href": {
-												"type": "string",
-												"example": "https://api.github.com/repos/Codertocat/Hello-World/pulls/2/commits"
-											}
-										}
-									},
-									"statuses": {
-										"type": "object",
-										"properties": {
-											"href": {
-												"type": "string",
-												"example": "https://api.github.com/repos/Codertocat/Hello-World/statuses/ec26c3e57ca3a959ca5aad62de7213c562f8c821"
-											}
-										}
-									}
-								}
-							},
-							"author_association": {
-								"type": "string",
-								"example": "OWNER"
-							}
-						}
-					},
-					"head": {
-						"type": "object",
-						"properties": {
-							"label": {
-								"type": "string",
-								"example": "Codertocat:changes"
-							},
-							"ref": {
-								"type": "string",
-								"example": "changes"
-							},
-							"sha": {
-								"type": "string",
-								"example": "ec26c3e57ca3a959ca5aad62de7213c562f8c821"
-							},
-							"user": {
-								"$ref": "#/components/schemas/user"
-							},
-							"repo": {
-								"$ref": "#/components/schemas/repository"
-							}
-						}
-					},
-					"base": {
-						"type": "object",
-						"properties": {
-							"label": {
-								"type": "string",
-								"example": "Codertocat:master"
-							},
-							"ref": {
-								"type": "string",
-								"example": "master"
-							},
-							"sha": {
-								"type": "string",
-								"example": "f95f852bd8fca8fcc58a9a2d6c842781e32a215e"
-							},
-							"user": {
-								"$ref": "#/components/schemas/user"
-							},
-							"repo": {
-								"$ref": "#/components/schemas/repository"
-							}
-						}
-					}
-				}
-			}
-		}`),
-	},
+//Builder is an interface containing all necessary functions required to build an ServiceDetails structure
+type Builder interface {
+	BuildServiceDetails() (ServiceDetails, error)
+	GetApplicationRegistryURL() string
 }
 
-var url = "http://application-registry-external-api.kyma-integration.svc.cluster.local:8081/" + os.Getenv("GITHUB_CONNECTOR_NAME") + "-app/v1/metadata/services"
+//ServiceRegister is an interface containing all necessary functions required to register service in Kyma's application registry
+type ServiceRegister interface {
+	RegisterService() (string, apperrors.AppError)
+}
+
+type serviceRegister struct {
+	envName      string
+	builder      Builder
+	register     ServiceRegister
+	retryDelay   int
+	retriesCount int
+}
+
+//NewServiceRegister creates a serviceRegister instance with the passed in interface
+func NewServiceRegister(deploymentEnvName string, b Builder, retryTime int, retries int) serviceRegister {
```

Comment:
I would improve naming a bit. The main responsibility of this component is communication with Application Registry. It is able to perform a request and interpret a response. So it is a _client_. I propose to rename ServiceRegister to ApplicationRegistryClient.